ansible/library/port_alias.py

```diff
@@ -184,12 +249,15 @@ def main():
             except Exception, e:
                 num_asic = 1
 
-
+        switchid = 0
         for asic_id in range(num_asic):
+            if asic_id is not None:
+                switchid = start_switchid + asic_id
             if num_asic == 1:
                 asic_id = None
-            (aliases_asic, portmap_asic, aliasmap_asic, portspeed_asic, front_panel_asic, asicifnames_asic)\
-                = allmap.get_portmap(asic_id, m_args['include_internal'])
+            (aliases_asic, portmap_asic, aliasmap_asic, portspeed_asic, front_panel_asic, asicifnames_asic,
+             sysport_asic) = allmap.get_portmap(asic_id, m_args['include_internal'],
+                                                hostname=m_args['hostname'], switchid=switchid)
```

Comment:
hostname is not a required argument, should check if hostname exists before using it.

Reply:
passing include_internal is not my change. but i can fix that too.